### PR TITLE
Tasks 2-4 completed

### PR DESCRIPTION
**Pull request by vovkandrew**

### CODE REVIEW

File: src/main/java/amazonreviewsapp/springboot/controller/ParseController.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+import amazonreviewsapp.springboot.parser.CSVFileParser;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class
+ParseController {
```
Comment (dmytro-hryhoruk) [inline src/main/java/amazonreviewsapp/springboot/controller/ParseController.java]:
```suggestion
public class ParseController {
```

File: src/main/java/amazonreviewsapp/springboot/controller/ParseController.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    private static final String PATH = "src/main/java/amazonreviewsapp/springboot/Reviews.csv";
```
Comment (dmytro-hryhoruk) [inline src/main/java/amazonreviewsapp/springboot/controller/ParseController.java]: it's better to add this file in resources. This way you will be able to get it with only single name.

File: src/main/java/amazonreviewsapp/springboot/model/User.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+import javax.persistence.FetchType;
+import javax.persistence.Id;
+import javax.persistence.ManyToMany;
+import javax.persistence.OneToMany;
+import javax.persistence.Table;
+import java.util.Set;
+
+@Entity
+@Table(name = "users")
+public class User {
```
Comment (dmytro-hryhoruk) [inline src/main/java/amazonreviewsapp/springboot/model/User.java:14]: entity creating should be in a separate PR.

File: src/main/java/amazonreviewsapp/springboot/model/User.java [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    @Id
+    private String userId;
+    private String profileName;
+    private String profilePassword;
+    @ManyToMany(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
```
Comment (dmytro-hryhoruk) [inline src/main/java/amazonreviewsapp/springboot/model/User.java]: if one user will be deleted it may delete all roles related to him. Other user may have those roles too so let's not  use CascadeType.ALL

File: src/main/java/amazonreviewsapp/springboot/parser/CSVFileParser.java [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+@Component
+public class CSVFileParser {
+    private static final int TIMESTAMP_MULTIPLIER = 1000;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private ReviewRepository reviewRepository;
```
Comment (dmytro-hryhoruk) [inline src/main/java/amazonreviewsapp/springboot/parser/CSVFileParser.java]: Let's create a services for this repositories and use them in other services.

File: src/main/java/amazonreviewsapp/springboot/parser/CSVFileParser.java [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+        } catch (CsvValidationException | IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private User getUserFromReader(String[] line) {
+        User user = new User();
+        user.setUserId(line[2]);
+        user.setProfileName(line[3]);
+        user.setProfilePassword(line[3]);
```
Comment (dmytro-hryhoruk) [inline src/main/java/amazonreviewsapp/springboot/parser/CSVFileParser.java]: Don't use magic numbers. Create an informative constant for it

File: src/main/java/amazonreviewsapp/springboot/parser/CSVFileParser.java [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+        review.setHelpDenum(Long.parseLong(line[5]));
+        review.setScore(Long.parseLong(line[6]));
+        review.setTime(getTimeFromTimestamp(line[7]));
+        review.setReviewSummary(line[8]);
+        review.setText(line[9]);
+        return review;
+    }
+
+    private String getTimeFromTimestamp(String time) {
+        Timestamp timestamp = new Timestamp(Long.parseLong(time)*TIMESTAMP_MULTIPLIER);
```
Comment (dmytro-hryhoruk) [inline src/main/java/amazonreviewsapp/springboot/parser/CSVFileParser.java]: So, in the end did you manage to get the real time out of it? Could you please send the screenshot of an example in comment below.

File: src/main/java/amazonreviewsapp/springboot/controller/ParseController.java (hunk 2 above)
Comment (dmytro-hryhoruk) [reply src/main/java/amazonreviewsapp/springboot/controller/ParseController.java]: after all let's better remove this constant from controller. Path to the file should not be hardcoded in your application.

File: src/main/java/amazonreviewsapp/springboot/model/User.java (hunk 3 above)
Comment (dmytro-hryhoruk) [inline src/main/java/amazonreviewsapp/springboot/model/User.java:14]: Is this the user from reviews or that the one that will be using your app?

**Review by dmytro-hryhoruk — COMMENTED** [review]
Let's add tests for your services, you can use constants there

File: src/main/java/amazonreviewsapp/springboot/controller/ParseController.java (hunk 2 above)
Comment (vovkandrew) [reply src/main/java/amazonreviewsapp/springboot/controller/ParseController.java]: Sorry, but, unfortunately, I can't get where should I move the file pass, will join you on the QA to clarify that

File: src/main/java/amazonreviewsapp/springboot/model/User.java (hunk 3 above)
Comment (vovkandrew) [reply src/main/java/amazonreviewsapp/springboot/model/User.java:14]: I thought Bohdan said that we can join several prs, anyway, what's the difference?

File: src/main/java/amazonreviewsapp/springboot/parser/CSVFileParser.java (hunk 7 above)
Comment (vovkandrew) [reply src/main/java/amazonreviewsapp/springboot/parser/CSVFileParser.java]: image

Everything works :)

File: src/main/java/amazonreviewsapp/springboot/controller/ParseController.java (hunk 2 above)
Comment (vovkandrew) [reply src/main/java/amazonreviewsapp/springboot/controller/ParseController.java]: Direct path to the .csv file at the repositories folder added

File: src/main/java/amazonreviewsapp/springboot/model/User.java (hunk 4 above)
Comment (vovkandrew) [reply src/main/java/amazonreviewsapp/springboot/model/User.java]: CascadeType changed to Persist

File: src/main/java/amazonreviewsapp/springboot/parser/CSVFileParser.java (hunk 5 above)
Comment (vovkandrew) [reply src/main/java/amazonreviewsapp/springboot/parser/CSVFileParser.java]: An additional layer of services for each repository interface added

File: src/main/java/amazonreviewsapp/springboot/parser/CSVFileParser.java (hunk 6 above)
Comment (vovkandrew) [reply src/main/java/amazonreviewsapp/springboot/parser/CSVFileParser.java]: All magic numbers were replaced with constants

File: src/main/java/amazonreviewsapp/springboot/model/User.java (hunk 3 above)
Comment (vovkandrew) [reply src/main/java/amazonreviewsapp/springboot/model/User.java:14]: Each user entity has its own password which is a duplicate of their profile name, will test that later when will add a security layer

File: src/main/java/amazonreviewsapp/springboot/parser/CSVFileParser.java (hunk 7 above)
Comment (vovkandrew) [reply src/main/java/amazonreviewsapp/springboot/parser/CSVFileParser.java]: Btw, I tested that with a random timestamp from the .csv file with reviews

File: src/main/java/amazonreviewsapp/springboot/service/ReviewService.java [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+@Service
+public class ReviewService {
+    @Autowired
+    private ReviewRepository reviewRepository;
+
+    public void save(Review review) {
+        reviewRepository.save(review);
+    }
+
+
```
Comment (dmytro-hryhoruk) [inline src/main/java/amazonreviewsapp/springboot/service/ReviewService.java:17]:
```suggestion
```

**Review by dmytro-hryhoruk — APPROVED** [review]
